### PR TITLE
feat: unify internal transfer transactions

### PR DESCRIPTION
**Pull request by euharrison**
Create the hook `useTransfer` which can be used to any Namada(internal) transfer transaction (not ibc, maybe yet?).

Then, apply it to the `<NamadaTransfer>`, `<MaspShield>` and `<MaspUnshield>`.

Please note that this is the first step for https://github.com/anoma/namada-interface/issues/1297 as the notification implementation for `Transfer` and `Masp` page will be the same now, but they both still has some notification issues that should be handled in a different PR focused on the notification systems, not the sdk transaction

### CODE REVIEW

File: apps/namadillo/src/App/Masp/MaspShield.tsx [hunk 1]
```diff
@@ -127,38 +115,30 @@ export const MaspShield: React.FC = () => {
...
+      if (txResponse) {
+        const txList = createTransferDataFromNamada(
+          txKind,
+          selectedAsset.asset,
+          rpcUrl,
+          txResponse
+        );
+
+        // Currently we don't have the option of batching transfer transactions
+        if (txList.length === 0) {
```
Comment (pedrorezende) [inline apps/namadillo/src/App/Masp/MaspShield.tsx:129]: Nitpick: let's prefer the use of the invariant? :)

File: apps/namadillo/src/App/Masp/MaspShield.tsx [hunk 2]
```diff
@@ -127,38 +115,30 @@ export const MaspShield: React.FC = () => {
...
+        const tx = txList[0];
+        setTransaction(tx);
+        storeTransaction(tx);
+      } else {
+        throw "Invalid transaction response";
+      }
     } catch (err) {
       setGeneralErrorMessage(err + "");
-      setCurrentStepIndex(0);
+      setTransaction(undefined);
```
Comment (pedrorezende) [inline apps/namadillo/src/App/Masp/MaspShield.tsx:141]: After talking to @ChrisHoltDesign we won't have the timeline active on the "Signing" anymore, which simplifies A LOT of things. Maybe you can redirect directly to the transactions page after the `storeTransaction` just like I did on #1454  - we can do it in a later PR too

File: apps/namadillo/src/App/NamadaTransfer/NamadaTransfer.tsx [hunk 3]
```diff
@@ -196,21 +123,10 @@ export const NamadaTransfer: React.FC = () => {
...
-
-      if (!selectedAsset) {
-        throw new Error("No asset is selected");
-      }
-
-      if (typeof gasConfig === "undefined") {
-        throw new Error("No gas config");
-      }
+      invariant(sourceAddress, "Source address is not defined");
+      invariant(chainId, "Chain ID is undefined");
```
Comment (pedrorezende) [inline apps/namadillo/src/App/NamadaTransfer/NamadaTransfer.tsx:127]: Maybe, just maybe, we could use a function to validate these variables here? We're repeating this validation multiple times

**Review by pedrorezende — APPROVED** [review]
Awesome PR!!! 🚀  Super neat!

File: apps/namadillo/src/App/Masp/MaspShield.tsx (hunk 2 above)
Comment (euharrison) [reply apps/namadillo/src/App/Masp/MaspShield.tsx:141]: Oh, this is a copy and paste from <NamadaTransfer> that you implemented 😁 

I'll open a new PR focused on the Timeline 👍

File: apps/namadillo/src/App/NamadaTransfer/NamadaTransfer.tsx (hunk 3 above)
Comment (euharrison) [reply apps/namadillo/src/App/NamadaTransfer/NamadaTransfer.tsx:127]: Yeah, I agree, some variables on the begin of the component as well. I'll explore this more on the next PRs

**Comment by euharrison** [comment]
I'll merge it and address the comments in the next PR :)

**Pull request merged**